CMakeLists.txt

```diff
@@ -70,6 +72,43 @@ macro(apply_llvm_compile_flags target)
     target_compile_definitions(${target} PRIVATE ${LLVM_DEFINITIONS})
 endmacro()
 
+# Check if using external LLVM/MLIR install tree
+if (DEFINED MLIR_DIR)
+    if (DEFINED MLIR_DIR)
+        set(MLIR_EXT_USE_LLVM_INSTALL_TREE ON)
+    else()
+        message(FATAL_ERROR "Need to define both LLVM_DIR and MLIR_DIR or none.")
+    endif()
+elseif (DEFINED LLVM_DIR)
+    message(FATAL_ERROR "Need to define both LLVM_DIR and MLIR_DIR or none.")
+else()
+    set(MLIR_EXT_USE_LLVM_INSTALL_TREE OFF)
+endif()
+
+if(GPU_ENABLE)
+    # Check if using external level-zero
+    if(DEFINED LEVEL_ZERO_DIR)
+        set(MLIR_EXT_USE_EXT_LEVEL_ZERO ON)
```

Comment:
Let us move to `IMEX` as the prefix of all our CMake flags.

Reply:
OK